presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileWriter.java

```diff
@@ -188,11 +188,15 @@ private void writeMetadataProperty(String key, String value)
     public void close()
             throws IOException
     {
-        writeRowGroup();
-        output.close();
-        keySectionOutput.destroy();
-        for (ColumnEncoder columnEncoder : columnEncoders) {
-            columnEncoder.destroy();
+        try {
+            writeRowGroup();
+            output.close();
+        }
+        finally {
+            keySectionOutput.destroy();
+            for (ColumnEncoder columnEncoder : columnEncoders) {
+                columnEncoder.destroy();
```

Comment:
Can destroy throw? If so, you need to handle exceptions for each one so that a single failure doesn't cause the others to leak.

Also, given that a failure here would result in a leak, this is probably something we should log as WARN.